deploy/rp-production-global-subscription.json

```diff
@@ -23,6 +29,42 @@
                     "[subscription().id]"
                 ]
             },
+            "condition": "[parameters('fullDeploy')]",
+            "apiVersion": "2018-01-01-preview"
+        },
+        {
+            "name": "0e2c3a2e-c25e-43bc-8fcd-4d9788a89d2c",
+            "type": "Microsoft.Authorization/roleDefinitions",
+            "properties": {
+                "roleName": "ARO v4 Extended Virtual Machine Contributor",
+                "permissions": [
+                    {
+                        "actions": [
+                            "Microsoft.Compute/virtualMachineScaleSets/*",
+                            "Microsoft.Network/loadBalancers/backendAddressPools/join/action",
+                            "Microsoft.Network/loadBalancers/probes/join/action",
+                            "Microsoft.Network/loadBalancers/read",
+                            "Microsoft.Network/locations/*",
+                            "Microsoft.Network/networkInterfaces/*",
+                            "Microsoft.Network/networkSecurityGroups/join/action",
+                            "Microsoft.Network/networkSecurityGroups/read",
+                            "Microsoft.Network/publicIPAddresses/join/action",
+                            "Microsoft.Network/publicIPAddresses/read",
+                            "Microsoft.Network/virtualNetworks/read",
+                            "Microsoft.Network/virtualNetworks/subnets/join/action",
+                            "Microsoft.Resources/deployments/*",
+                            "Microsoft.Resources/subscriptions/resourceGroups/read",
+                            "Microsoft.ManagedIdentity/userAssignedIdentities/assign/action",
+                            "Microsoft.ManagedIdentity/userAssignedIdentities/read",
+                            "Microsoft.Network/dnsZones/read"
+                        ]
+                    }
+                ],
+                "assignableScopes": [
+                    "[subscription().id]"
+                ]
+            },
+            "condition": "[parameters('fullDeploy')]",
             "apiVersion": "2018-01-01-preview"
         }
```

Comment:
Clearly, the deployer tool is going to need at minimum "Read and Assign User Assigned Identity" as well as "Virtual Machine Contributor"; so be it.  Let's not create a new definition - it's a chicken and egg problem.

As far as "Microsoft.Network/dnsZones/read", give up on using Outputs for this and have a separate API call made only on fullDeploy?

Reply:
What is your suggestion here? Do not create roles and use 2 default ones[1]? Or create something else? 


[1] https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#virtual-machine-contributor and https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#managed-identity-operator

Reply:
the granting of roles to the deployer is external to the deploy process, so don't worry about it.